### PR TITLE
8197859: VS2017 Complains about UINTPTR_MAX definition in globalDefinitions_VisCPP.hpp

### PR DESCRIPTION
**Pull request by apavlyutkin**
Hi! Please review another backport from MSVS2019 seria. This one fixes type declarations made by globalDefinitions_VisCPP.hpp. The patch from 11u applied with the following changes:

- inttypes.h is included conditionally under `_MSC_VER >= 1800` because the header was introduced only in MSVS 2013 but we have to keep support of the earlier MSVS versions 
- the duplicates of declarations made in inttypes.h are not just removed but quoted with `_MSC_VER < 1800`
- INT64_C & UINT64_C declarations removed cuz they duplicate ones done by `stdint.h`

Verification: 2019 build (both 32/64) now fails with 
```
ad_x86_64_pipeline.obj : error LNK2011: precompiled object not linked in; image may not run
jvm.dll : fatal error LNK1120: 1 unresolved externals
```
error (to be fixed by backport of 8043492)

Regression: 2017/2013/2012/2010 full build - ok

@kimbarrett @dholmes-ora if you took a look at that it would be very much appreciated

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8197859](https://bugs.openjdk.org/browse/JDK-8197859): VS2017 Complains about UINTPTR_MAX definition in globalDefinitions_VisCPP.hpp

### Reviewers
 * [Paul Hohensee](https://openjdk.org/census#phh) (@phohensee - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk8u-dev pull/77/head:pull/77` \
`$ git checkout pull/77`

Update a local copy of the PR: \
`$ git checkout pull/77` \
`$ git pull https://git.openjdk.org/jdk8u-dev pull/77/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 77`

View PR using the GUI difftool: \
`$ git pr show -t 77`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk8u-dev/pull/77.diff">https://git.openjdk.org/jdk8u-dev/pull/77.diff</a>

</details>

### CODE REVIEW

**Comment by apavlyutkin** [comment]
ping

**Comment by apavlyutkin** [comment]
Guys, this one stops a seria of backports required to support VS 2019/22. Could somebody to review it? Thank you

**Comment by apavlyutkin** [comment]
ping

**Comment by apavlyutkin** [comment]
ping

**Review by jerboaa — CHANGES_REQUESTED** [review]
Why is `generated-configure.sh` part of this change? It doesn't touch m4 scripts.

**Comment by jerboaa** [comment]
> Why is `generated-configure.sh` part of this change? It doesn't touch m4 scripts.

Ah, you mentioned it in the description. Forgotten in https://github.com/openjdk/jdk8u-dev/pull/33. Either way, this shouldn't be part of this backport. It's unrelated and just confuses people when looking at the backport after the fact. Please open and 8u-specific bug and correct this. Also, it's not clear why `generated-configure.sh` would be this large. Are you generating it with the right system? If you do it on a wrong one it can get messy and mixes unrelated changes due to version skew.

**Comment by apavlyutkin** [comment]
Ok, I removed generated-configure.sh from the PR, probably the best option here to commit it after the rest of VS2019 backports (there are 2-3 more) are done

**Comment by apavlyutkin** [comment]
ping

File: hotspot/src/share/vm/utilities/globalDefinitions_visCPP.hpp [hunk 1]
```diff
@@ -77,53 +81,19 @@
...
 #define NULL_WORD NULL
 
-// Some MS Visual Studio versions do not seem to have INT64_C and UINT64_C
-// even with __STDC_CONSTANT_MACROS defined.
-#ifndef INT64_C
-#define INT64_C(c)  (c ## i64)
-#endif
-#ifndef UINT64_C
-#define UINT64_C(c) (c ## ui64)
-#endif
```
Comment (jerboaa) [inline hotspot/src/share/vm/utilities/globalDefinitions_visCPP.hpp:87]: These removals aren't in the original changeset. See JDK-8272714 and JDK-8272214. I guess we should conditionalize on the VS version in use if that's a problem?

**Review by jerboaa — COMMENTED** [review]
It would be good if some Windows build person could review this patch as well.

**Comment by apavlyutkin** [comment]
The definitions now are done by including of `stdint.h` that is surely available even in msvs 2010 (as I remember a far earlier). I have no idea why they were declared localy. Looks like I forgot to mention this deviation from the original changeset.

**Comment by jerboaa** [comment]
> The definitions now are done by including of `stdint.h` that is surely available even in msvs 2010 (as I remember a far earlier). I have no idea why they were declared localy. Looks like I forgot to mention this deviation from the original changeset.

OK, thanks.

**Comment by apavlyutkin** [comment]
> > The definitions now are done by including of `stdint.h` that is surely available even in msvs 2010 (as I remember a far earlier). I have no idea why they were declared localy. Looks like I forgot to mention this deviation from the original changeset.
> 
> OK, thanks.

Luckily the guys from my old job still use msvs-2008. I asked them to check presense of `stdint.h`, and the header was introduced exactly in msvs-2010. Looks like jdk-8 didn't use the header to provide compatibility with the compilers older than 2010, but now it does not make a sense

**Review by phohensee — CHANGES_REQUESTED** [review]
The copyright date should stay at 2018 because that's what the original commit had. Other than that, lgtm.

**Comment by apavlyutkin** [comment]
> The copyright date should stay at 2018 because that's what the original commit had. Other than that, lgtm.

fixed

**Review by phohensee — APPROVED** [review]
Thanks, lgtm.

**Comment by apavlyutkin** [comment]
Thank you for the review

**Pull request closed without merging**